lua/octo/utils.lua

```diff
@@ -190,22 +190,28 @@ function M.commit_exists(commit, cb)
     :start()
 end
 
+local last_get_file_job = nil
+
 function M.get_file_at_commit(path, commit, cb)
   if not Job then
     return
   end
-  Job
-    :new({
-      enable_recording = true,
-      command = "git",
-      args = { "show", string.format("%s:%s", commit, path) },
-      on_exit = vim.schedule_wrap(function(j_self, _, _)
-        local output = table.concat(j_self:result(), "\n")
-        local stderr = table.concat(j_self:stderr_result(), "\n")
-        cb(vim.split(output, "\n"), vim.split(stderr, "\n"))
-      end),
-    })
-    :start()
+  local job = Job:new {
+    enable_recording = true,
+    command = "git",
+    args = { "show", string.format("%s:%s", commit, path) },
+    on_exit = vim.schedule_wrap(function(j_self, _, _)
+      local output = table.concat(j_self:result(), "\n")
+      local stderr = table.concat(j_self:stderr_result(), "\n")
+      cb(vim.split(output, "\n"), vim.split(stderr, "\n"))
+    end),
+  }
+  if last_get_file_job ~= nil then
+    last_get_file_job:and_then(job)
+  else
+    job:start()
+  end
+  last_get_file_job = job
```

Comment:
this should, make each job run sequentially, ideally there is some kind of concurrency parameter on Job, so that we can run more than one at a time...

Reply:
Nice, have you noticed any performance impact?

Reply:
I have a PR with a hundred of file changes, this is quite fast to run, I haven't done any measures, but it should not be  a noticeable changes for most PR that usually have less changes than that